### PR TITLE
Port ccloudvm to cobra

### PR DESCRIPTION
**Pull request by rbradford**
Refactor ccloudvm internals in order to support the port over to cobra command line framework and then port.

### CODE REVIEW

File: ccvm/ccvm.go [hunk 1]
```diff
@@ -170,41 +170,57 @@ func StartFlags() (VMSpec, error) {
...
+		return nil, nil, err
 	}
 
 	ws.Mounts = in.Mounts
 	ws.Hostname = wkld.spec.Hostname
 	if ws.NoProxy != "" {
 		ws.NoProxy = fmt.Sprintf("%s,%s", ws.Hostname, ws.NoProxy)
-	} else if ws.HTTPProxy != "" || ws.HTTPSProxy != "" {
-		ws.NoProxy = ws.Hostname
+		ws.PackageUpgrade = "false"
```
Comment (markdryan) [inline ccvm/ccvm.go]: It looks like there's been some sort of merge issue here.  This block of code is duplicated after the if statement.

Also, there's a change in behaviour here.  Previously, if no_proxy was not defined but http_proxy or https_proxy were, ccloudvm was initialising ws.NoProxy to ws.Hostname.  The code no longer does this.

File: cmd/root.go [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+// Execute is the entry into the cmd package from the main package.
+func Execute() {
+	if err := rootCmd.Execute(); err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+}
+
+func init() {
```
Comment (markdryan) [inline cmd/root.go]: Is there something missing here is this function not needed?

**Comment by rbradford** [comment]
All done!

**Comment by markdryan** [comment]
Actually, I've got another problem

```
$ ccloudvm start --cpus=2 --mem=2
Booting VM with 7 GB RAM and 4 cpus
VM Started
```

File: cmd/start.go [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+		return ccvm.Start(ctx, &startSpec)
+	},
+}
+
+func init() {
+	rootCmd.AddCommand(startCmd)
+
+	var flags flag.FlagSet
+	ccvm.VMFlags(&flags, &createSpec)
```
Comment (markdryan) [inline cmd/start.go]: Presumably it's this.  I guess we should be passing &startSpec as the last parameter.

**Comment by markdryan** [comment]
Also, the default package-upgrade seems to have changed.  Previously, it was set effectively to false.  Setting it to true might be the correct thing to do but this will slow down the CI builds for ciao and CC.

**Comment by markdryan** [comment]
@rbradford I had the fixes locally so I just pushed them to your branch.  If you're okay with them I'll merge.

**Comment by rbradford** [comment]
These changes look good. Thank you @markdryan for your diligent testing!

**Pull request merged**